src/contact.rs

```diff
@@ -768,6 +779,9 @@ impl Contact {
             return &self.name;
         }
         if !self.authname.is_empty() {
+            // normally, name is initialized by authname and this condition would not be needed.
```

Comment:
I don't understand this comment. `authname` can be empty because we received an email where `From` contains only the email address. It can always happen, so this check should never be removed.

Reply:
authnames also always initialize the name (modulo bugs)

if authname is set, also name is set.
so, if the first condition fails, (no name set), the second condition will fails as well.

maybe we can remove/change the comment, the second condition currently displays the authname instead of the email address, so, an edit done can be reverted this way. maybe this is better than just displaying the email-address as core-c did.

Reply:
Why do we initialize the name without user manually doing anything?

As discussed in #1014, `name` should only be updated by explicit user action. So I would rather remove this comment and then, in a separate PR probably, remove all code changing `name` without an explicit user action. And then in a DB migration reset all `name`s that are the same as `authname` to empty strings, only once, just to fix this bug.

Reply:
`name` should be always set to allow easy searching. but anyway, we can remove the comment as it targets exactly the case where the name is empty. i do a pr for that.